### PR TITLE
Personal messages

### PR DESCRIPTION
**Pull request by dlesnef1**
Added personal messages to backend

### CODE REVIEW

File: src/main/java/com/ithaca/message/MessageServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    @Autowired
+    ThreadRepository threadRepository;
+
+    @Override
+    public Thread findThread(Long userId, String otherUserName) {
+        User user = userRepository.findOne(userId);
+        User otherUser = userRepository.findByName(otherUserName);
+
+        if (user == null || otherUser == null || user == otherUser) {
```
Comment (NSBPittman) [inline src/main/java/com/ithaca/message/MessageServiceImpl.java:31]: So I can't message myself NOOOOOO

File: src/main/java/com/ithaca/message/MessageServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            user.getThreads().add(currentThread);
+            recipient.getThreads().add(currentThread);
+        }
+
+        Message message = new Message(user, currentThread, text,
+                new SimpleDateFormat("yyyy-MM-dd HH:mm:ss").format(new java.util.Date()));
+        user.getMessages().add(message);
+        currentThread.getMessages().add(message);
+
+        userRepository.save(user);
```
Comment (NSBPittman) [inline src/main/java/com/ithaca/message/MessageServiceImpl.java:74]: explain these 2 lines

Comment (dlesnef1) [inline src/main/java/com/ithaca/message/MessageServiceImpl.java:74]: Which two? All of the saving stuff?

File: src/main/java/com/ithaca/user/User.java [hunk 3]
```diff
@@ -27,19 +33,28 @@
...
+        this.threads = new ArrayList<>();
     }
 
     public User(String name, String password, String securityQuestion, String securityAnswer) {
         this.name = name;
         this.password = password;
         this.securityQuestion = securityQuestion;
         this.securityAnswer = securityAnswer;
+        this.messages = new ArrayList<>();
     }
```
Comment (NSBPittman) [inline src/main/java/com/ithaca/user/User.java:59]: does Create account not need to initiate threads?

File: src/main/java/com/ithaca/message/MessageServiceImpl.java (hunk 1 above)
Comment (dlesnef1) [inline src/main/java/com/ithaca/message/MessageServiceImpl.java:31]: Might want to rethink this, people without friends will be heartbroken.

File: src/main/java/com/ithaca/message/MessageServiceImpl.java (hunk 2 above)
Comment (NSBPittman) [inline src/main/java/com/ithaca/message/MessageServiceImpl.java:74]: userRepository.save(user/recipient)

does userRepository just save who is in the thread or??

File: src/main/java/com/ithaca/message/Message.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
+package com.ithaca.message;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.ithaca.user.User;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+
+/**
+ * Created by David on 4/4/16.
```
Comment (NSBPittman) [inline src/main/java/com/ithaca/message/Message.java:10]: Shelby and Chris should be looking through this... They should be looking at your code so they have an idea whats happening, and to learn about code reviews.

File: src/main/java/com/ithaca/user/User.java (hunk 3 above)
Comment (dlesnef1) [inline src/main/java/com/ithaca/user/User.java:59]: SEE THIS IS WHY CODE REVIEWS ARE GOOD

File: src/main/java/com/ithaca/message/Message.java (hunk 4 above)
Comment (ShelbyCohen) [inline src/main/java/com/ithaca/message/Message.java:10]: I have been looking through it. Thanks though nick.

**Pull request merged**